src/internal/m365/onedrive/collection.go

```diff
@@ -531,11 +534,29 @@ func (oc *Collection) populateDriveItem(
 		// attempts to read bytes.  Assumption is that kopia will check things
 		// like file modtimes before attempting to read.
 		itemReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
-			itemData, err := oc.getDriveItemContent(ctx, oc.driveID, item, errs)
+			rc, err := oc.getDriveItemContent(ctx, oc.driveID, item, errs)
 			if err != nil {
 				return nil, err
 			}
 
+			itemData := rc
+
+			// Add per item extensions if available
+			if len(itemExtensionFactory) > 0 {
+				extRc, extInfo, err := extensions.AddItemExtensions(
+					ctx,
+					rc,
+					itemInfo,
+					itemExtensionFactory)
+
+				if err != nil {
+					return nil, clues.Stack(err).WithClues(ctx)
+				}
+
+				itemData = extRc
+				itemInfo.OneDrive.Extension = extInfo
```

Comment:
@ryanfkeepers Quick question for you: Since sharepoint libraries also utilize onedrive collections, this code is currently failing for sharepoint backups with nil reference. 

To fix it, I have a few ideas. Wanted to run them by you quickly :) 

1. Currently, `Collection` doesn't distinguish between OneDrive vs Sharepoint library items. Add `path.ServiceType` and/or `path.Category` to onedrive `Collections` & `Collection`. Use that here to decide between filling `Sharepoint.Extension` vs `OneDrive.Extension`. 
2. Expose `[Extension](https://github.com/alcionai/corso/pull/3750/commits/98a3f3abb330a64c9618845dbf3864b805a41049#diff-47f6ad08905a9164dd5eeedd907a7559345c803d7775970e88e9121a0f50a033R991)` directly in `ItemInfo` rather than inside `ItemInfo.OneDrive` or `ItemInfo.Sharepoint`

I am leaning towards 1, since 2 mangles scope of nested structures. Also, the extension is tied to the item. Ideally, if it's a onedrive item, everything should be under `ItemInfo.OneDrive` scope.

Reply:
I'd say `2`.  The reason being that:

> Ideally, if it's a onedrive item, everything should be under ItemInfo.OneDrive scope

Is slightly off the mark.  If it's a onedrive item, all item data that is _specific to onedrive_ should be under ItemInfo.OneDrive.  Data that can generically represent any item can be promoted to the top level ItemInfo.  

Extensions are not specific to onedrive.  They're just starting their implementation in onedrive.  So it makes sense to promote them.

Reply:
Hmm. This loose coupling was not sitting well with me. If we do 2),  ideally the extension should have some identifier to say that this is a OneDrive extension vs Sharepoint vs ... But for adding that identifier, we'd need 1) again from my previous comment. 

Let me go with 2). It will push some burden on to the ItemInfo consumer to associate extension with OneDrive/Sharepoint, but benefit here is that we can keep onedrive.`Collection` generic.